### PR TITLE
UIREC-59 Use only select Location plugin

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UIREC-59

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/TitleDetails/AddPieceModal/AddPieceModal.js [hunk 1]
```diff
@@ -150,15 +146,13 @@ const AddPieceModal = ({
         </Row>
         <Row>
           <Col xs>
-            <FieldSelectFinal
-              dataOptions={locations}
-              disabled
-              fullWidth
-              label={<FormattedMessage id="ui-receiving.piece.location" />}
+            <FieldLocationFinal
+              id={locationId}
```
Comment (aliaksei-chumakou) [inline src/TitleDetails/AddPieceModal/AddPieceModal.js]: bad naming, it's not clear why do you set `id` prop, that usually id in DOM as location id.

File: src/TitleReceive/TitleReceiveList.js [hunk 2]
```diff
@@ -9,78 +9,66 @@ import {
...
+    callNumber: record => (
+      <Field
+        name={`${field}[${record.rowIndex}].callNumber`}
+        component={TextField}
+        marginBottom0
+        fullWidth
+      />
+    ),
+    location: record => (
+      <div>
```
Comment (aliaksei-chumakou) [inline src/TitleReceive/TitleReceiveList.js]: what's the purpose of that `div`?

File: src/TitleReceive/TitleReceiveList.js [hunk 3]
```diff
@@ -9,78 +9,66 @@ import {
...
 } from '@folio/stripes-acq-components';
 
 import { PIECE_FORMAT_LABELS } from '../common/constants';
 
 const visibleColumns = ['checked', 'barcode', 'format', 'hasRequest', 'comments', 'location', 'itemStatus', 'callNumber'];
 
-export const TitleReceiveList = ({ fields, props: { locationOptions, selectLocation, toggleCheckedAll } }) => {
+export const TitleReceiveList = ({ fields, props: { selectLocation, toggleCheckedAll } }) => {
   const field = fields.name;
-  const cellFormatters = useMemo(
```
Comment (aliaksei-chumakou) [inline src/TitleReceive/TitleReceiveList.js]: you removed `useMemo`, I wonder if there a reason for that?

Comment (AndreiShumski2) [reply src/TitleReceive/TitleReceiveList.js]: Previously it was based on `locationOptions`

File: src/TitleReceive/TitleReceiveList.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/TitleReceive/TitleReceiveList.js]: Removed it

File: src/TitleReceive/TitleReceiveList.js (hunk 3 above)
Comment (aliaksei-chumakou) [reply src/TitleReceive/TitleReceiveList.js]: yes, but now it's re-created every time

**Pull request merged**